gensim/downloader.py

```diff
@@ -0,0 +1,359 @@
+    data_folder_dir = os.path.join(base_dir, data_)
+    data = info()
+    corpora = data['corpus']
+    models = data['model']
+    if data_ not in corpora and data_ not in models:
+        raise Exception(
+            "Incorect Model/corpus name. Use info() or"
+            " python -m gensim.downloader -c to get a list of models/corpora"
+            " available.")
+    compressed_folder_name = "{f}.tar.gz".format(f=data_)
+    compressed_folder_dir = os.path.join(base_dir, compressed_folder_name)
+    if get_data_status(data_) != "downloaded":
+        if not os.path.exists(data_folder_dir):
+            logger.info("Creating %s", data_folder_dir)
+            os.makedirs(data_folder_dir)
+            if os.path.exists(data_folder_dir):
+                logger.info("Creation of %s successful.", data_folder_dir)
+            else:
+                raise Exception(
+                    "Not able to create {a}. Make sure you have the correct read/"
+                    "write permissions for {b} or you can try creating it manually".
+                    format(a=data_folder_dir, b=base_dir))
+        logger.info("Downloading %s", data_)
+        urllib.urlretrieve(url, compressed_folder_dir)
+        data_url = data_links(data_)
+        if data_url is not None:
+            index = data_url.rfind("/")
+            data_dir = os.path.join(data_folder_dir, data_url[index + 1:])
+            urllib.urlretrieve(data_url, data_dir)
+        if calculate_md5_checksum(data_folder_dir, compressed_folder_dir) == get_checksum(data_, "download"):
+            logger.info("%s downloaded", data_)
+            update_data_log_file(data_, status="downloaded")
+        else:
+            logger.error("There was a problem in downloading the data. Retrying.")
+            _download(data_)
+
+    if get_data_status(data_) != "installed":
+            tar = tarfile.open(compressed_folder_dir)
+            logger.info("Extracting files from %s", data_folder_dir)
+            tar.extractall(data_folder_dir)
+            tar.close()
+            if calculate_md5_checksum(data_folder_dir) == get_checksum(data_, "installation"):
+                update_data_log_file(data_, status="installed")
+                logger.info("%s installed", data_)
+            else:
+                logger.error("There was a problem in installing the dataset/model. Retrying.")
+                _download(data_)
+
+
+def get_filename(data_):
+    """Function of retrieving the filename of corpus/model.
+
+    Args:
+        data_(string): Name of the corpus/model.
+
+    Returns:
+        string: Returns the filename of the model/corpus.
+    """
+    data = info()
+    corpora = data['corpus']
+    models = data['model']
+    if data_ in corpora:
+        return data['corpus'][data_]["filename"]
+    elif data_ in models:
+        return data['model'][data_]["filename"]
+
+
+def load(data_, return_path=False):
+    """Loads the corpus/model to the memory, if return_path is False.
```

Comment:
Code style: Python docstrings use imperative mode ("Load X", not "Loads X"). Here and elsewhere.